fia_api/scripts/transforms/loq_transform.py

```diff
@@ -0,0 +1,67 @@
+"""
+Module provides the LoqTransform class, an implementation of the Transform abstract base class for LOQ instrument
+scripts.
+"""
+
+import logging
+
+from db.data_models import Job
+from sqlalchemy import ColumnElement
+from sqlalchemy.dialects.postgresql import JSONB
+
+from fia_api.scripts.pre_script import PreScript
+from fia_api.scripts.transforms.transform import Transform
+
+logger = logging.getLogger(__name__)
+
+
+class LoqTransform(Transform):
+    """
+    LoqTransform applies modifications to LOQ instrument scripts based on reduction input parameters in a Job
+    entity.
+    """
+
+    def apply(self, script: PreScript, job: Job) -> None:  # noqa: C901
+        logger.info("Beginning LOQ transform for job %s...", job.id)
+        lines = script.value.splitlines()
+        # MyPY does not believe ColumnElement[JSONB] is indexable, despite JSONB implementing the Indexable mixin
```

Comment:
Looked into this, can't you use the `astext` method to avoid adding all these type ignores?

Source: https://stackoverflow.com/a/29975187

Reply:
No, because we go on to start indexing it, we don't want a string but an addressable JSONB object.

We could do `json.dumps(jsonb.astext())` but at that point we are casting twice (just to get rid of a MyPy issue, as it is MyPy that is incorrect not the code) whilst trying to return as soon as reasonable. We do this is multiple other places in the code base.

Reply:
Well if it is the case that it's a MyPy issue then can't be remedied then.